### PR TITLE
Feature/core 8477 calendar closing when clicking in dropdown

### PR DESCRIPTION
**Pull request by asanzdj**
Jira tasks:

- https://geoblink.atlassian.net/browse/CORE-8477
- https://geoblink.atlassian.net/browse/CORE-8300

### CODE REVIEW

File: src/elements/GeoDropdown/GeoDropdown.vue [hunk 1]
```diff
@@ -423,6 +426,23 @@ export default {
...
+      const POPUP_REF_NAME = this.POPUP_REF_NAME
+
+      function checkForPopup (refs) {
+        return _.reduce(refs, (acc, ref, refName) => {
+          if (acc) return acc
+
+          if (refName === POPUP_REF_NAME) {
+            if (ref === $event.target || ref.contains($event.target)) return true
+          }
+          return checkForPopup(_.get(ref, '$refs'))
```
Comment (asanzdj) [inline src/elements/GeoDropdown/GeoDropdown.vue:440]: This is needed to the last ref to check

File: src/elements/GeoCalendar/GeoCalendarRoot.mixin.js [hunk 2]
```diff
@@ -159,6 +159,14 @@ export default {
     popupClass: {
       type: [String, Array, Object],
       required: false
+    },
+
+    /**
+     * Should appear the calendar opened by default
+     */
+    openedByDefault: {
```
Comment (Navas23) [inline src/elements/GeoCalendar/GeoCalendarRoot.mixin.js:167]: Why do we need this? Is this a required improvement?

Comment (asanzdj) [reply src/elements/GeoCalendar/GeoCalendarRoot.mixin.js:167]: No, I forgot about this 🤦‍♀️

File: src/elements/GeoDropdown/GeoDropdown.vue [hunk 3]
```diff
@@ -423,6 +426,23 @@ export default {
...
+        }, false)
+      }
+
+      if (checkForPopup(popupContentRefs)) return
```
Comment (fofi) [inline src/elements/GeoDropdown/GeoDropdown.vue:444]: I think we could change the name of this method to something more descriptive

What do you think about:
* checkClickOnReferencesPopup
* checkClickOnPopup
* hasClickOnChildrenPopup

File: src/elements/GeoDropdown/GeoDropdown.vue [hunk 4]
```diff
@@ -423,6 +426,23 @@ export default {
         return
       }
 
+      const popupContentRefs = _.get(this.$slots.popupContent, '[0].context.$refs')
+
+      const POPUP_REF_NAME = this.POPUP_REF_NAME
+
+      function checkForPopup (refs) {
```
Comment (fofi) [inline src/elements/GeoDropdown/GeoDropdown.vue:433]: As this is an auxiliary method we could move it to the end of this function, so we make it more readable

**Review by fofi — COMMENTED** [review]
Apart from the comments looks great

**Review by fofi — APPROVED** [review]

**Review by Navas23 — APPROVED** [review]
If you tried this in the app lets go!

**Review by fofi — APPROVED** [review]

**Review by Navas23 — APPROVED** [review]

**Review by fofi — APPROVED** [review]

**Pull request merged**